src/main/java/ann/parser/Parser.java

```diff
@@ -0,0 +1,201 @@
+package ann.parser;
+
+import ann.commands.*;
+import ann.data.InputPattern;
+import ann.data.tasks.TaskType;
+
+/**
+ * Represents a parser to parse user input into appropriate commands.
+ *
+ * @author Hong Anh
+ * @version 0.1
+ */
+public class Parser {
+
+    /**
+     * Returns a Command which is the result of parsing the user input.
+     *
+     * @param input a String which is the user input.
+     * @return a Command.
+     */
+    public static Command parse(String input) {
+        if(input.toLowerCase().equals("bye")) {
+            return new ExitCommand();
+        } else if (input.toLowerCase().equals("list")){
+            return new ListCommand();
+        } else if (input.length() < 3) {
+            return new IncorrectCommand("Oops! Please enter a valid command!");
+        } else if(input.substring(0, 3).toLowerCase().equals("add")) {
+            return handleAdd(input.substring(3));
+        } else if(input.length() < 4) {
+            return new IncorrectCommand("Oops! Please enter a valid command!");
+        } else if(input.substring(0, 4).toLowerCase().equals("find")) {
+            return handleFind(input.substring(4));
+        } else if(input.substring(0, 4).toLowerCase().equals("mark")) {
+            return handleMark(input.substring(4));
+        } else if(input.length() < 6) {
+            return new IncorrectCommand("Oops! Please enter a valid command!");
+        } else if(input.substring(0, 6).toLowerCase().equals("unmark")) {
+            return handleUnmark(input.substring(6));
+        } else if(input.substring(0, 6).toLowerCase().equals("delete")) {
+            return handleDelete(input.substring(6));
+        } else {
+            return new IncorrectCommand("Oops! Please enter a valid command!");
+        }
+    }
+
+    /**
+     * Parses the contents of an 'add' command and returns the appropriate Command object.
+     *
+     * @param input a String which represents the contents of the 'add' command.
+     * @return an appropriate Command object.
+     */
+    private static Command handleAdd(String input) {
+        if(input.isBlank()) {
+            return new IncorrectCommand("Oops! Please specify the task you wish to add!");
+        } else if(input.charAt(0) != ' ') {
```

Comment:
The coding standard suggests a space after an `if` or `else if`!  It seems that this is followed in some places but not elsewhere, so maybe you could have overlooked it.